### PR TITLE
docs: rewrite pull query docs page

### PR DESCRIPTION
**Pull request by cprasad1**
### Description 
Rewrite pull query docs page https://docs.ksqldb.io/en/latest/developer-guide/ksqldb-reference/select-pull-query/

### Testing done 
_Describe the testing strategy. Unit and integration tests are expected for any behavior changes._

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: docs/developer-guide/ksqldb-reference/select-pull-query.md
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/select-pull-query.md:47]:
```suggestion
statement. For example, you can create a table named `GRADES` by using a [CREATE TABLE](../../ksqldb-reference/create-table) 

```

File: docs/developer-guide/ksqldb-reference/select-pull-query.md
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/select-pull-query.md:61]:
```suggestion
You can fetch the current state of the `TOP_TEN_RANKS` materialized view by using a pull query:

```

File: docs/developer-guide/ksqldb-reference/select-pull-query.md
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/select-pull-query.md:70]:
```suggestion
If the `pageviews_by_region` materialized view is windowed:

```

File: docs/developer-guide/ksqldb-reference/select-pull-query.md
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/select-pull-query.md:118]:
```suggestion
You can fetch the current state of the `INNER_JOIN` materialized view by using a pull query:

```

**Review by JimGalasyn — APPROVED** [review]
LGTM, with a few suggestions! If you'd like this to go live now, it needs to be cherry-picked to the `0.17.0-ksqldb` branch, and to keep branches in sync, also to the `0.18.x-ksqldb` branch. #LifeWithoutPintMerge

**Review by vvcephei — COMMENTED** [review]
Thanks, I agree with @agavra 's feedback. Otherwise, LGTM!

File: docs/developer-guide/ksqldb-reference/select-pull-query.md
Comment (AlanConfluent) [inline docs/developer-guide/ksqldb-reference/select-pull-query.md:39]: We should clarify this in more detail for when table scans are enabled.  We should say, after the "Time and WIndows" link, that enabling table scans removes many of the restrictions and allows most queries, though it'll be less performant, depending on the size of the underlying view.

File: docs/developer-guide/ksqldb-reference/select-pull-query.md
Comment (AlanConfluent) [reply docs/developer-guide/ksqldb-reference/select-pull-query.md:47]: I think we should have two sections: `Queryable views` and `WHERE clause guidelines`.  They could be before the examples and just summarize each.

`Queryable views` would state Almog's assertion like:
Any table created with `AS SELECT` can be queried.  Some examples include:
- Tables created by querying another table
- Tables created with a group by clause
- Tables created by joining other tables.

`WHERE clause guidelines` would state that by default, only key lookups are enabled.  These have the following requirements:
- Key column(s) must use an equality comparison to a literal (e.g. `KEY = 'abc'`)
- On windowed tables, `WINDOWSTART` and `WINDOWEND` can be optionally compared to literals.

To enable table scans, you set `ksql.query.pull.table.scan.enabled=true`.  Once they're are enabled, additional expressions are allowed:
- Key column(s) using range comparisons to literals
- Non key columns to be used alone, without key references
- Columns to be compared to other columns
- References to subsets of columns from a multi-column key

Optionally, these can be worked into the examples, but I think it would be useful to summarize.  Also should probably add some more examples that hit some of the above, especially table scans.

**Review by AlanConfluent — APPROVED** [review]

File: docs/developer-guide/ksqldb-reference/select-pull-query.md
Comment (AlanConfluent) [inline docs/developer-guide/ksqldb-reference/select-pull-query.md:57]: Forgot to add this:
- Complex expressions without direct column references (e.g. `instr(NAME_COL, 'hello') > 0`)

Comment (cprasad1) [reply docs/developer-guide/ksqldb-reference/select-pull-query.md:61]: added

File: docs/developer-guide/ksqldb-reference/select-pull-query.md
Comment (agavra) [inline docs/developer-guide/ksqldb-reference/select-pull-query.md:52]: we might want to say what a "table scan" is (perhaps the note below is a good place for this). @JimGalasyn thoughts?

File: docs/developer-guide/ksqldb-reference/select-pull-query.md
Comment (agavra) [inline docs/developer-guide/ksqldb-reference/select-pull-query.md:56]:
```suggestion
`ksql.query.pull.table.scan.enabled=true`. Once table scans are enabled, the following 

```

File: docs/developer-guide/ksqldb-reference/select-pull-query.md
Comment (agavra) [inline docs/developer-guide/ksqldb-reference/select-pull-query.md:61]: do we also support UDFs and function calls? Might make sense to call out what is _not_ supported as opposed to what is here.

File: docs/developer-guide/ksqldb-reference/select-pull-query.md
Comment (agavra) [inline docs/developer-guide/ksqldb-reference/select-pull-query.md:102]: I'm not entirely sure this example is helpful (the windowed one below is helpful because it shows the usage of window bounds on key lookups)

**Review by agavra — APPROVED** [review]
Left some comments inline - thanks for the PR!

File: docs/developer-guide/ksqldb-reference/select-pull-query.md
Comment (cprasad1) [reply docs/developer-guide/ksqldb-reference/select-pull-query.md:102]: We have stated in our 0.17 blog and 0.17 docs explicitly that Pull queries are not supported on Table-Table joins. This just clarifies to a certain degree that they are now supported.

Comment (agavra) [reply docs/developer-guide/ksqldb-reference/select-pull-query.md:102]: IMO I think it's sufficiently clarified above when you discuss that any CTAS table can be queried. Not a big deal though, we can keep it or get another opinion on it.

Comment (cprasad1) [reply docs/developer-guide/ksqldb-reference/select-pull-query.md:102]: Let's keep it. It is also one of the examples in our 0.18 blog

File: docs/developer-guide/ksqldb-reference/select-pull-query.md
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/select-pull-query.md:59]:
```suggestion
-   Non-key columns to be used alone, without key references.

```

File: docs/developer-guide/ksqldb-reference/select-pull-query.md
Comment (AlanConfluent) [reply docs/developer-guide/ksqldb-reference/select-pull-query.md:61]: Yeah, we do support UDFs and function calls.  Really, anything goes.  The below line gives an example of this: `(e.g. instr(NAME_COL, 'hello') > 0)`

File: docs/developer-guide/ksqldb-reference/select-pull-query.md
Comment (cprasad1) [reply docs/developer-guide/ksqldb-reference/select-pull-query.md]: ewmoved the commentary on `SELECT`. It was a bit confusing and wrong

File: docs/developer-guide/ksqldb-reference/select-pull-query.md
Comment (AlanConfluent) [reply docs/developer-guide/ksqldb-reference/select-pull-query.md:32]: It streams back a chunked response.  Technically, it's a single response, though maybe it's a little misleading, since I think that wording was intended to mean it's returned in one fell swoop (one chunk!).  Maybe we want to change the description to either say "Streams back a chunked response as rows are fetched" or just make it a little more vague so that it's a little less tied to an implementation by saying "the API responds with resulting rows".

**Comment by cprasad1** [comment]
master is broken. merging this in as it is just docs

**Pull request merged**
